### PR TITLE
fix when awareness is disabled, always changing

### PR DESCRIPTION
**Pull request by sanarena**
## what
When we disable zone awareness, Terraform want to put in availability_zone_count on every single run,
so resource is always marked as changed.
This pull request fix this issue.

## why
Avoid unnecessary AWS calls and changes.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -152,7 +152,7 @@ resource "aws_elasticsearch_domain" "default" {
     warm_type                = var.warm_enabled ? var.warm_type : null
 
     dynamic "zone_awareness_config" {
-      for_each = var.availability_zone_count > 1 ? [true] : []
+      for_each = var.availability_zone_count > 1 && var.zone_awareness_enabled == true ? [true] : []
```
Comment (nitrocode) [inline main.tf:155]:
```suggestion
      for_each = var.availability_zone_count > 1 && var.zone_awareness_enabled ? [true] : []
```

**Comment by sanarena** [comment]
@jhosteny @milldr can help with reviewing and merging please? thanks

**Review by Gowiem — APPROVED** [review]

**Comment by Gowiem** [comment]
@sanarena Thanks for the contribution! Released as https://github.com/cloudposse/terraform-aws-elasticsearch/releases/tag/0.35.1

**Pull request merged**